stmhal/usb.c

```diff
@@ -205,6 +206,32 @@ STATIC mp_obj_t pyb_usb_mode(mp_uint_t n_args, const mp_obj_t *pos_args, mp_map_
         { MP_QSTR_hid, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = (mp_obj_t)&pyb_usb_hid_mouse_obj} },
     };
 
+    // fetch the current usb mode -> pyb.usb_mode()
+    if (n_args == 0) {
+        uint8_t mode = -1;
```

Comment:
You can simplify this code here, since the mode variable is not needed if in host mode.  And the big switch below is also not needed in host mode.